docs/flow/config-language/syntax.md

```diff
@@ -6,3 +6,135 @@ weight: 200
 ---
 
 # Syntax
+The River syntax is designed to be easy to read and write. Essentially, there
+are just two high-level elements to it; _Attributes_ and  _Blocks_. 
+
+The main purpose of River is to define components for the Grafana Agent Flow
+mode, so that they can be wired together to form programmable telemetry
+pipelines.
+As such, the ordering of blocks and attributes within the River configuration
+file is not important; the language will consider all direct and indirect
+dependencies between elements to determine the order which must be used for the
+evaluation.
+
+## Attributes and Blocks
+
+### Attributes
+_Attributes_ are used to configure individual settings. They always take the
+form of `<ATTRIBUTE_NAME> = <ATTRIBUTE_VALUE>`. They can appear either as
+top-level elements or nested within blocks.
+
+The following sets the `log_level` attribute to `"debug"`.
+
+```river
+log_level = "debug"
+```
+
+All `<ATTRIBUTE_NAME>`s must be valid River identifiers: they must be made up
+of one or more UTF-8 letters, digits or underscores, but cannot start with a
+digit.
+
+The `<ATTRIBUTE_VALUE>` can be either a constant value of a valid River
+[type]({{< relref "./expressions/types_and_values.md" >}}) (eg. string,
+boolean, number) or an [_expression_]({{< relref "./expressions/_index.md" >}})
+that computes and continuously re-evaluates more complex attribute values.
+
+### Blocks
+_Blocks_ are used to configure components by grouping any number of attributes
+or nested blocks using curly braces.
+
+Blocks have a _name_ (the type of the component it defines), an
+optional _label_ and a body that contains any number of arguments or nested
+unlabeled blocks. Labels are used to disambiguate between multiple top-level
+blocks of the same name.
+
+```
+// Pattern for creating a block with an optional user-label:
+<BLOCK NAME> ["<BLOCK LABEL>"] {
+	// Block body can contain attributes and nested unlabeled blocks
+	<IDENTIFIER> = <EXPRESSION> // Attribute
+
+	<NESTED_BLOCK_NAME> {
+		// Nested block body
+	}
+}
+```
+
+The `<BLOCK_NAME>` must correspond to a registered Flow [component]({{< relref "./components.md" >}}).
+
+If the optional `<BLOCK_LABEL>` is set, it has to be a valid River identifier:
+consisting of one or more UTF-8 letters, digits or underscores, but it cannot
+begin with a digit.
```

Comment:
Rather than giving the rules for a valid identifier multiple times, can we have a dedicated header for defining what an "identifier" is?

Reply:
Yes, it was there in the first place, putting it back now.